jenkins/bootstrap.py

```diff
@@ -88,6 +88,9 @@ def call(cmd, stdin=None, check=True, output=None):
                     reads.remove(fdesc)
                     continue
                 logging.warning(line[:-1])
+        if not ret[0]:
+            if proc.poll() is not None:
+                break  # process exited without closing pipes (timeout?)
```

Comment:
Sot he third arg in select.select() is for exceptional circumstances... does this include not closing the pipes?